src/userSync.js

```diff
@@ -0,0 +1,169 @@
+import * as utils from 'src/utils';
+import { config } from 'src/config';
+
+export function newUserSync(userSyncConfig) {
```

Comment:
This looks really good... but there's one small tweak I'd like to propose:

```
import { config } from 'src/config';

const cookiesAreSupported = !utils.isSafariBrowser() && utils.cookiesAreEnabled();
export function newUserSync(userSyncDependencies) { ... }

export const userSync = newUserSync({
  config: config.getConfig('userSync'),
  cookiesAreSupported: cookiesAreSupported
});
```

It solves a testability problem you described in `userSync_spec` (see the comment there), and also makes this module fit into the larger Prebid architecture described in #1510.